src/test/java/org/junit/platform/engine/test/event/ExecutionEventConditions.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2015-2016 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v1.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.junit.platform.engine.test.event;
+
+import static java.util.function.Predicate.isEqual;
+import static org.assertj.core.api.Assertions.allOf;
+import static org.assertj.core.data.Index.atIndex;
+import static org.junit.platform.commons.util.FunctionUtils.where;
+import static org.junit.platform.engine.TestExecutionResult.Status.ABORTED;
+import static org.junit.platform.engine.TestExecutionResult.Status.FAILED;
+import static org.junit.platform.engine.TestExecutionResult.Status.SUCCESSFUL;
+import static org.junit.platform.engine.test.event.ExecutionEvent.Type.DYNAMIC_TEST_REGISTERED;
+import static org.junit.platform.engine.test.event.ExecutionEvent.Type.FINISHED;
+import static org.junit.platform.engine.test.event.ExecutionEvent.Type.SKIPPED;
+import static org.junit.platform.engine.test.event.ExecutionEvent.Type.STARTED;
+import static org.junit.platform.engine.test.event.ExecutionEvent.byPayload;
+import static org.junit.platform.engine.test.event.ExecutionEvent.byTestDescriptor;
+import static org.junit.platform.engine.test.event.ExecutionEvent.byType;
+import static org.junit.platform.engine.test.event.TestExecutionResultConditions.cause;
+import static org.junit.platform.engine.test.event.TestExecutionResultConditions.status;
+
+import java.util.List;
+
+import org.assertj.core.api.Assertions;
+import org.assertj.core.api.Condition;
+import org.assertj.core.api.SoftAssertions;
+import org.junit.platform.engine.TestDescriptor;
+import org.junit.platform.engine.TestExecutionResult;
+import org.junit.platform.engine.TestExecutionResult.Status;
+import org.junit.platform.engine.support.descriptor.EngineDescriptor;
+import org.junit.platform.engine.test.event.ExecutionEvent.Type;
+
+/**
+ * Collection of AssertJ conditions for {@link ExecutionEvent}.
+ *
+ * @since 1.0
+ */
+public class ExecutionEventConditions {
+
+	@SafeVarargs
+	public static void assertRecordedExecutionEventsContainsExactly(List<ExecutionEvent> executionEvents,
```

Comment:
It's not clear to me if it makes sense for either `executionEvents` or `conditions` to be empty.

If they're not meant to be empty, I'd suggest throwing an `IllegalArgumentException` or similar if `executionsEvents.isEmpty()`, and changing

```
Condition<? super ExecutionEvent>... conditions
```

to

```
Condition<? super ExecutionEvent> firstCondition, Condition<? super ExecutionEvent>... rest
```